### ISSUE TITLE
Async texture upload fails after some time

### ISSUE THREAD

**Issue by marpies** [Gamua/Adobe-Runtime-Support#32]
## Problem Description

On iOS, BitmapData fails to be uploaded asynchronously after a number of previously successful uploads. The app will either crash or continue to run without any textures showing on the screen. If the app continues to run and the app is minimized and re-launched then a crash occurs with the following stack trace:

```
0   libGPUSupportMercury.dylib    	0x2be92742 gpus_ReturnUnexpectedKillClient + 2
1   libGPUSupportMercury.dylib    	0x2be93202 gpusSubmitDataBuffers + 118
2   libGPUSupportMercury.dylib    	0x2be93078 gldCreateContext + 204
3   GLEngine                      	0x2769ce06 gliCreateContextWithShared + 598
4   OpenGLES                      	0x27779a8c -[EAGLContext commonInitWithAPI:properties:] + 404
5   OpenGLES                      	0x277798c6 -[EAGLContext initWithAPI:sharegroup:] + 110
6   Async Upload Sample           	0x00029874 0xe000 + 112756
7   Async Upload Sample           	0x006d7e40 0xe000 + 7118400
8   Async Upload Sample           	0x004b1c48 0xe000 + 4865096
9   Async Upload Sample           	0x004b1d70 0xe000 + 4865392
10  Async Upload Sample           	0x004b1928 0xe000 + 4864296
11  libsystem_pthread.dylib       	0x33bcfddc _pthread_body + 136
12  libsystem_pthread.dylib       	0x33bcfd4e _pthread_start + 114
13  libsystem_pthread.dylib       	0x33bcdaf8 thread_start + 4
```

Tested with different AIR version from 25 to the current beta (28.0.0.120).

[Link to Adobe Tracker](https://tracker.adobe.com/#/view/AIR-4198527)

## Steps to Reproduce

(See attached `sample-project.zip`)

Upload a BitmapData asynchronously, wait for the upload to complete and display the texture. Dispose the texture after a small delay. Repeat this process. The sample project uploads 4 textures to reach the crash point faster.

It is easier to reproduce on older, less powerful iOS device (iPad 2nd or 3rd generation, iPhone 4S...). In the provided sample, the upload starts failing after about 2000 uploads have been made (at least on the aforementioned devices).

## Known Workarounds

Use synchronous texture upload.

[sample-project.zip](https://github.com/Gamua/Adobe-Runtime-Support/files/1538855/sample-project.zip)

**Comment by PrimaryFeather** [comment]
Thanks a lot for taking so much time to make this reproducible, Marcel! I'll test this myself, too, and will send Adobe a heads-up.

**Comment by marpies** [comment]
Looks like this will never get fixed, Adobe just went silent on this issue. They've made some attempts to fix it, but the issue never went away and it is no longer listed under "known issues". It can still be reproduced with AIR 30.0.0.96 beta.

**Comment by gevorg-gsar** [comment]
I think Adobe does very poor job at fixing really important issues with AIR SDK. There are many features and fixes in every SDK release that have absolutely low value for the mobile game devs, yet thats what they chose to work on.
@PrimaryFeather i think that all devs will be happy if you have a serious talk with AIR SDK team!

**Comment by marpies** [comment]
Yeah I sometimes wonder how are they making decisions on what to fix and implement. It's true this issue haven't received too many votes from the community, but Adobe has already put some effort into fixing it so it's really strange they decided to stop any further work. It's quite sad because the async upload is such a great feature.

**Comment by PrimaryFeather** [comment]
You're right, it seems this issue dropped from their radar again. I'll try to get this to their attention again!
